lib/utils.js

```diff
@@ -26,7 +35,8 @@ function isArray(val) {
  * @returns {boolean} True if value is a Buffer, otherwise false
  */
 function isBuffer(val) {
-  return ![undefined, null].includes(val) && val.constructor === Buffer;
+  if (val === null || isUndefined(val) || val.constructor === null ||  isUndefined(val.constructor))  return false;
+  return typeof val.constructor.isBuffer === 'function' && val.constructor.isBuffer(val);
```

Comment:
@Alanscut Thanks for reminding me lint rules. Then can we write all conditions in the same statement? And you have an extra space before `isUndefined` and `return false` now. :P

```
return val !== null && !isUndefined(val) && val.constructor !== null && !isUndefined(val.constructor)
    && typeof val.constructor.isBuffer === 'function' && val.constructor.isBuffer(val);
```

Reply:
agree, this will be more clear!